arkouda/categorical.py

```diff
@@ -1356,3 +1359,51 @@ def parse_hdf_categoricals(
                 result_categoricals[base_name] = Categorical.from_codes(**cat_parts)
 
         return removal_names, result_categoricals
+
+    def send_array(self, hostname: str, port: int_scalars):
+        """
+        Sends a Categorical object to a different Arkouda server
+
+        Parameters
+        ----------
+        hostname : str
+            The hostname where the Arkouda server intended to
+            receive the Categorical is running.
+        port : int_scalars
+            The port to send the array over. This needs to be an
+            open port (i.e., not one that the Arkouda server is
+            running on). This will open up `numLocales` ports,
+            each of which in succession, so will use ports of the
+            range {port..(port+numLocales)} (e.g., running an
+            Arkouda server of 4 nodes, port 1234 is passed as
+            `port`, Arkouda will use ports 1234, 1235, 1236,
+            and 1237 to send the array data).
+            This port much match the port passed to the call to
+            `ak.receive_array()`.
+
+
+        Returns
+        -------
+        A message indicating a complete transfer
+
+        Raises
+        ------
+        ValueError
+            Raised if the op is not within the pdarray.BinOps set
+        TypeError
+            Raised if other is not a pdarray or the pdarray.dtype is not
+            a supported dtype
+        """
+        # hostname is the hostname to send to
+        args = {
+            "codes": self.codes,
+            "categories": self.categories,
+            "objType": "categorical",
+            "NA_codes": self._akNAcode,
```

Comment:
Based on the `__init__`, it looks like there are cases where this can be an `int`. Maybe I just haven't seen it yet, but are we checking that somewhere? If not, it would probably be best to create the `pdarray` here.

Again, this is definitely not checked in other locations in existing code. I will add an issue to look into this and make sure it gets updated.